### PR TITLE
feat: deploy metamask

### PR DESCRIPTION
**Pull request by asiaziola**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -39,6 +39,7 @@
     "pg-query-stream": "^4.2.3",
     "warp-contracts": "1.2.13",
     "warp-contracts-pubsub": "^1.0.3",
+    "warp-signature": "1.0.2",
```
Comment (asiaziola) [inline package.json:42]: can be installed only after upgrading `warp-signature` to 1.0.3 (still waiting for the lib publication)

File: src/gateway/router/routes/deployContractRoute.ts [hunk 2]
```diff
@@ -1,34 +1,60 @@
...
-import {BUNDLR_NODE2_URL} from '../../../constants';
-import {uploadToBundlr} from './sequencerRoute';
-import {updateCache} from "../../updateCache";
+import { GQLTagInterface, SmartWeaveTags } from 'warp-contracts';
+import { evalType } from '../../tasks/contractsMetadata';
+import { getCachedNetworkData } from '../../tasks/networkInfoCache';
+import { BUNDLR_NODE2_URL } from '../../../constants';
+import { uploadToBundlr } from './sequencerRoute';
+import { updateCache } from '../../updateCache';
+// @ts-ignore
```
Comment (asiaziola) [inline src/gateway/router/routes/deployContractRoute.ts:10]: can be deleted after upgrading typescript and setting `moduleResolution` to `nodenext`, current ts version does not support this property. some other compilation errors will need to be fixed after ts upgrade so i decided to create a separate pr for this one. probably together with linter addition :))

File: package.json (hunk 1 above)
Comment (ppedziwiatr) [reply package.json:42]: btw. please move it to the new plugins monnorepo. Just remember, that for now the whole monorepo share the same tagging scheme, so the versioning must be "global" for all of the plugins.

Also, remember to set the warp-contracts in the plugin itself as a peerDependency.

File: src/gateway/router/routes/deployContractRoute.ts (hunk 2 above)
Comment (ppedziwiatr) [reply src/gateway/router/routes/deployContractRoute.ts:10]: Please create a separate issue for this task.

Comment (ppedziwiatr) [reply src/gateway/router/routes/deployContractRoute.ts:10]: also, remember that last time you've used the linter in the gateway code - some raw sql queries were fucked up :-)

File: src/gateway/router/routes/deployContractRoute.ts [hunk 3]
```diff
@@ -124,29 +150,50 @@ function tagValue(name: string, tags: GQLTagInterface[]): string | undefined {
...
+    { name: 'Uploader-Bundler', value: BUNDLR_NODE2_URL },
     ...decodedTags,
   ];
 
-  return { tags, testnet };
+  return { tags, testnet, originalAddress, isEvmSigner };
+}
+
+async function verifyEvmSignature(tx: Transaction, logger: any) {
+  const signatureVerification = new EvmSignatureVerificationServerPlugin();
```
Comment (ppedziwiatr) [inline src/gateway/router/routes/deployContractRoute.ts:197]: do we need to create the plugin instance each time? does it store some state?

File: package.json (hunk 1 above)
Comment (asiaziola) [reply package.json]: done

File: src/gateway/router/routes/deployContractRoute.ts (hunk 3 above)
Comment (asiaziola) [reply src/gateway/router/routes/deployContractRoute.ts]: done.

File: package.json [hunk 4]
```diff
@@ -39,9 +39,10 @@
     "pg": "8.7.3",
     "pg-query-stream": "^4.2.3",
     "raw-body": "^2.5.1",
-    "warp-contracts": "1.2.13",
+    "warp-contracts": "1.2.24",
```
Comment (asiaziola) [inline package.json:42]: need to upgrade when this one is installed https://github.com/warp-contracts/warp/pull/257

**Pull request merged**